File: Python/String-2/count_hi.py
"""
Return the number of times that the string "hi" appears anywhere in the given string.

count_hi('abc hi ho') → 1
count_hi('ABChi hi') → 2
count_hi('hihi') → 2

"""

def count_hi(str):
  c=0
  for i in range(len(str)):
    if str[i:i+2]=="hi":
      c=c+1
  return c

File: Java/String-2/countHi.java
/*

Return the number of times that the string "hi" appears anywhere in the given string.

countHi("abc hi ho") ? 1
countHi("ABChi hi") ? 2
countHi("hihi") ? 2
*/

public int countHi(String str) {
  String comp="hi";
  int count=0;
  for(int i=0;i<str.length()-1;i++){
    if(comp.equals(str.substring(i,i+2)))
      count++;
  }
  return count;
}
